app/src/main/java/com/github/wnder/GuessLocationActivity.java

```diff
@@ -245,27 +262,40 @@ protected void onDestroy() {
      * Shows the real location of the picture
      */
     private void showActualLocation() {
-        //Update karma after a guess
-        if(!pictureID.equals(Picture.UNINITIALIZED_ID)){
-            ExistingPicture pic = new ExistingPicture(pictureID);
-            pic.addKarmaForGuess();
-        }
+        //TODO make the actual picture location appear
+        CameraPosition camPosition = new CameraPosition.Builder().target(cameraPosition).zoom(zoomFromKilometers(distance)).build();
+        mapboxMap.setCameraPosition(camPosition);
+    }
 
-        //Get real position
-        Point point = Point.fromLngLat(picturePosition.getLongitude(), picturePosition.getLatitude());
-        Polygon circle = TurfTransformation.circle(point, 200, "meters");
-        GeoJsonSource pictureSource = new GeoJsonSource(PICTURE_SOURCE_ID, circle);
+    private void drawCircle(LatLng position) {
+        Point center = Point.fromLngLat(position.getLongitude(), position.getLatitude());
+        Polygon outerCirclePolygon = TurfTransformation.circle(center,  distance + distance/15.0, "kilometers");
+        Polygon innerCirclePolygon = TurfTransformation.circle(center, (double) distance, "kilometers");
+
+        GeoJsonSource outerCircleSource = new GeoJsonSource(PICTURE_SOURCE_ID, outerCirclePolygon);
+
+        if (outerCircleSource != null) {
+            outerCircleSource.setGeoJson(Polygon.fromOuterInner(
+                    LineString.fromLngLats(TurfMeta.coordAll(outerCirclePolygon, false)),
+                    LineString.fromLngLats(TurfMeta.coordAll(innerCirclePolygon, false))
+            ));
+        }
 
-        //Set mapbox style
         Style style = mapboxMap.getStyle();
-        style.addSource(pictureSource);
+        style.addSource(outerCircleSource);
         style.addLayer(new FillLayer(PICTURE_LAYER_ID, PICTURE_SOURCE_ID).withProperties(
-                PropertyFactory.fillColor("#ff0000"),
+                PropertyFactory.fillColor(ContextCompat.getColor(this, R.color.red)),
                 PropertyFactory.fillOpacity(0.4f)
         ));
+    }
 
-        //Set camera position
-        CameraPosition position = new CameraPosition.Builder().target(picturePosition).zoom(14).build();
-        mapboxMap.setCameraPosition(position);
+    private double zoomFromKilometers(int kilometers) {
+        int absLat = Math.abs((int) cameraPosition.getLatitude());
+        double latDeformation = 0.00046*Math.pow(absLat, 2);
+        double offset = 13.6 - latDeformation;
```

Comment:
Magic numbers

Reply:
I'll add a comment but I didn't managed to find documentation for the latitude deformation on zoom.
If you find a better function let me know: https://docs.mapbox.com/help/glossary/zoom-level/